### PR TITLE
Set RELATED_IMAGE environment variables in operator deployment

### PR DESCRIPTION
**Pull request by andreasgerstmayr**
* add RELATED_IMAGE env vars to operator deployment
* remove images field from operator configuration

Resolves: #591

### CODE REVIEW

File: bundle/community/manifests/tempo-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -887,6 +896,15 @@ spec:
   maturity: alpha
   provider:
     name: Grafana Tempo Operator SIG
+  relatedImages:
```
Comment (andreasgerstmayr) [inline bundle/community/manifests/tempo-operator.clusterserviceversion.yaml:899]: :raised_hands:

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -38,6 +38,15 @@ spec:
         image: controller:latest
         args:
         - --leader-elect
+        env:
```
Comment (pavolloffay) [inline config/manager/manager.yaml:41]: It would be good to document how versions should be updated

Comment (andreasgerstmayr) [reply config/manager/manager.yaml:41]: I've added a section to `CONTRIBUTING.md`.

**Comment by rubenvp8510** [comment]
This looks good to me, I think we need to fix the conflicts first.

**Comment by andreasgerstmayr** [comment]
I've resolved the conflicts now

**Pull request merged**